routes/index.ts

```
@@ -266,17 +268,29 @@ const getCatch = async (req, res) => {
 
 const newCatch = async (req, res) => {
     if (req.headers['content-type'] == "application/xml") { stringParser(req); }
-    setTimeout( () => {
+    setTimeout(async () => {
         if (req.params.tripNum && req.body.tripNum && req.body.source && req.body.hauls) {
-                const newTrip = req.body;
-                newTrip.type = 'trips-api-catch';
-                newTrip.createdDate = moment().format();
-                masterDev.bulk({docs: [newTrip]}).then(
-                    () => {
-                        res.send('catch data saved');
-                        // catchEvaluator(req.params.tripNum);
-                    }
-                );
+            const newTrip = req.body;
+            newTrip.type = 'trips-api-catch';
+            newTrip.createdDate = moment().format();
+
+            const tripDocs = await masterDev.view('TripsApi', 'all_api_trips', { "key": req.params.tripNum });
+            if (tripDocs.rows.length === 0 ) {
+                res.status(500).send('tripNum does not exist');
+            } else {
+                const catchDocs = await masterDev.view('TripsApi', 'all_api_catch', { "key": req.params.tripNum, "include_docs": true });
+                const sourceTypes: string[] = jp.query(catchDocs, '$..source');
+                if (sourceTypes.includes(req.body.source)) {
+                    catchEvaluator(req.params.tripNum); // update results but don't add new catch doc
```

no need to do this right away, but the correct process here is to reject the newCatch request and stop processing it.
Should throw an error saying a logbook submission for trip number 100131 already exists, please submit updated data to tripCatch via put request to tripCatch/[tripNum]

---

ahhh okay got it, didn't realize there's another api for update. got it that makes sense then, I'll update it in the next review